Zpp/Mrp/MachineManagement/Machine.cs

```diff
@@ -26,8 +26,8 @@ public Id GetMachineGroupId()
 
         public override bool Equals(object obj)
         {
-            Machine other = (Machine) obj;
-            return _machine.Equals(other._machine);
+            var other = (Machine) obj;
+            return _machine.Equals(other?._machine);
         }
```

Comment:
Attention, are really wanted to add an questionmark after other? Why is this needed?
Second, please don't replace Types by var (the hard work by writing the type down, is allready done), this will make it less readable.

Reply:
Compiler, warning, possible null ref Exception. The question mark just supress the warning.
If you compare something with null it will throw at this point anyway.
Alternatively you can check for null and return false?

Reply:
The compiler warning should stay, else I won't find this mistake :). But you can fix it by null check and then return false as you suggested.

Reply:
Doublechecks for null values now.